src/masternodes/rpc_loan.cpp

```diff
@@ -643,6 +643,164 @@ UniValue listloanschemes(const JSONRPCRequest& request) {
+                    {"loanschemeid", RPCArg::Type::STR, RPCArg::Optional::OMITTED,
+                        "Unique identifier of the loan scheme (8 chars max). If empty, the default loan scheme will be selected (Optional)"
+                    },
+                    {"inputs", RPCArg::Type::ARR, RPCArg::Optional::OMITTED_NAMED_ARG,
+                        "A json array of json objects",
+                            {
+                                {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                {
+                                     {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                     {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                    },
+                },
+                RPCResult{
+                   "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+                },
+                RPCExamples{
+                   HelpExampleCli("createvault", "") +
+                   HelpExampleCli("createvault", "" "LOAN0001") +
+                   HelpExampleCli("createvault", "2MzfSNCkjgCbNLen14CYrVtwGomfDA5AGYv LOAN0001") +
+                   HelpExampleCli("createvault", "") +
+                   HelpExampleRpc("createvault", "\"\", LOAN0001")+
+                   HelpExampleRpc("createvault", "2MzfSNCkjgCbNLen14CYrVtwGomfDA5AGYv, LOAN0001")
+                },
+    }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload())
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Cannot create order while still in Initial Block Download");
+
+    pwallet->BlockUntilSyncedToCurrentChain();
+    LockedCoinsScopedGuard lcGuard(pwallet);
+
+    CVaultMessage vault;
+    if(request.params.size() > 0){
+        vault.ownerAddress = request.params[0].getValStr();
+        if(vault.ownerAddress.empty()){
+            // generate new address
+            LOCK(pwallet->cs_wallet);
+
+            if (!pwallet->CanGetAddresses()) {
+                throw JSONRPCError(RPC_WALLET_ERROR, "Error: This wallet has no available keys");
+            }
+            CTxDestination dest;
+            std::string error;
+            if (!pwallet->GetNewDestination(OutputType::LEGACY, "*", dest, error)) {
+                throw JSONRPCError(RPC_WALLET_KEYPOOL_RAN_OUT, error);
+            }
+            vault.ownerAddress = EncodeDestination(dest);
+        } else {
+            // check address validity
+            CTxDestination ownerDest = DecodeDestination(vault.ownerAddress);
+            if (!IsValidDestination(ownerDest)) {
+                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Error: Invalid owneraddress address");
+            }
+        }
+    }
+    vault.schemeId = pcustomcsview->GetDefaultLoanScheme().get();
```

Comment:
You should check if scheme exists

Reply:
this is done in mn_checks, if no loan scheme is given default loan scheme is assigned otherwise if the loan scheme is invalid it throws an error, should I change this behavior to do this check at rpc level?

Reply:
Yes, you can leave empty here. In mn_checks you should verify when `schemeId` is given it's a valid one.

Reply:
I see you are right, will push now.

Reply:
the check for the validity of the not empty loan scheme is done [here](https://github.com/DeFiCh/ain/blob/4bac34cb6eb5c9f9668c5c5081d07a93fedc1070/src/masternodes/mn_checks.cpp#L1835) is this hat you mean?